WTG.Analyzers.Test/RuleSetTest.cs

```diff
@@ -0,0 +1,43 @@
+using System;
+using System.IO;
+using System.Linq;
+using System.Xml.Linq;
+using Microsoft.CodeAnalysis;
+using NUnit.Framework;
+
+namespace WTG.Analyzers.Test
+{
+	[TestFixture]
+	class RuleSetTest
+	{
+		[Test]
+		public void WarnAll()
+		{
+			var rulesetPath = Path.Combine(TestContext.CurrentContext.TestDirectory, "..", "..", "WTG.Analyzers", "build", "WarnAll.ruleset");
+
+			var ruleSet = XElement.Load(rulesetPath)
+				.Element("Rules")
+				.Elements("Rule")
+				.ToDictionary(
+					x => x.Attribute("Id").Value,
+					x => (DiagnosticSeverity)Enum.Parse(typeof(DiagnosticSeverity), x.Attribute("Action").Value));
+
+			var descriptors = typeof(Rules).GetFields()
+				.Where(x => x.IsStatic && x.IsInitOnly && typeof(DiagnosticDescriptor).IsAssignableFrom(x.FieldType))
+				.Select(x => (DiagnosticDescriptor)x.GetValue(null));
+
+			Assert.Multiple(() =>
+			{
+				foreach (var descriptor in descriptors)
+				{
+					if (!ruleSet.TryGetValue(descriptor.Id, out var severity))
+					{
+						severity = descriptor.DefaultSeverity;
+					}
+
+					Assert.That(severity, Is.EqualTo(DiagnosticSeverity.Warning), descriptor.Id);
```

Comment:
Should we also have a test that the ruleset only contains rules we have defined, or is that too over-the-top?

Reply:
I thought about it and figured it wasn't worth it ... thinking about it again, it's a fairly trivial check.